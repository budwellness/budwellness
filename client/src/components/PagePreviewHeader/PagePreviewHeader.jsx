import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './PagePreviewHeader.module.scss';

function PagePreviewHeader({ title, text }) {
  return (
    <>
      <div className={styles.wrapper}>
        <div className={styles.container}>
          <h1 className={styles.title}>{title}</h1>
          <p className={styles.text}>{text}</p>
        </div>
      </div>
      <div className={styles.linkWpapper}>
        <NavLink to="/" className={styles.link}>
          Home &nbsp;/
        </NavLink>
        <span>{title}</span>
      </div>
    </>
  );
}

PagePreviewHeader.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
};

PagePreviewHeader.defaultProps = {
  title: 'Shop',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Inventore, blanditiis.',
};

export default PagePreviewHeader;
