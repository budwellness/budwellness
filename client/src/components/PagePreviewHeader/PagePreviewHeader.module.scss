@mixin for-size($size) {
  @media (min-width: $size) {
    @content;
  }
}

.wrapper {
  background-image: url(./images/pagePreviewHeader-bg.webp);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  text-align: center;
}

.container {
  max-width: 1380px;
}

.title {
  padding-top: 60px;
  margin-bottom: 15px;
  font-size: 36px;
  font-weight: 900;
  font-family: 'Nunito Sans', sans-serif;
  color: $main-green;
  @include for-size(768px) {
    font-size: 40px;
  }
  @include for-size(992px) {
    font-size: 56px;
    padding-top: 80px;
  }
}

.text {
  font-size: 17px;
  color: $main-text-color;
  line-height: 24px;
  max-width: 500px;
  margin: 0 auto;
  padding-bottom: 60px;
  @include for-size(992px) {
    padding-bottom: 90px;
  }
}

.linkWpapper {
  padding-top: 30px;
  font-size: 17px;
  font-family: 'Nunito Sans', sans-serif;
}

.link {
  color: $secondary-green;
  position: relative;
  padding-right: 8px;
  text-decoration: none;
}

.link::after {
  content: '';
  position: absolute;
  bottom: -12px;
  left: 0;
  right: 0;
  height: 2px;
  background-color: $secondary-green;
}

span {
  font-size: 17px;
  color: $main-text-color;
}
