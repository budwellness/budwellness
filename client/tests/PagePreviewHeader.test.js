/* eslint-disable */
import React from 'react';
import { render } from '@testing-library/react';
import PagePreviewHeader from '../src/components/PagePreviewHeader/PagePreviewHeader.jsx';
import { Router } from 'express';

describe('PagePreviewHeader testing snapshot', () => {
  test('should PagePreviewHeader match snapshot', () => {
    const { asFragment } = render(
      <Router>
        <PagePreviewHeader text="TITLE" />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
