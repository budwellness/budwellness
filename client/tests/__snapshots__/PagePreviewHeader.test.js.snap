// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePreviewHeader testing snapshot should PagePreviewHeader match snapshot 1`] = `<DocumentFragment />`;
